cfpb/owning-a-home
# Static email

Add static email form to OAH pages

## Additions

- Static email html, css, and js

## Testing

- Check out branch and run grunt  
- Check that email form displays correctly on all pages:
  - [ ] /explore-rates
  - [ ] /closing-disclosure
  - [ ] /loan-estimate
  - [ ] /process template
  - [ ] /loan-options
  - [ ] /loan-options/fha-loans
  - [ ] /loan-options/conventional-loans
  - [ ] /loan-options/special-loan-programs
  - [ ] /mortgage-estimate  

- Check form submission and messaging:
  - [ ] Error message appears when form submitted with missing email address (in Safari since Chrome shows its own tooltip message)
  - [ ] Error message appears when form submitted with incomplete email address (again in Safari)
  - [ ] Form submits without page reload when a full email address is entered. It will show an error message locally on submit (it also errors with JS turned off), but it should be successful on build.


## Review
- @sonnakim 

## Screenshots

__In brand footer:__
<img width="1309" alt="screen shot 2016-11-21 at 9 22 14 pm" src="https://cloud.githubusercontent.com/assets/778171/20512167/76448cc0-b032-11e6-98f8-d42828318ded.png">  



__In sidebar:__

<img width="1261" alt="screen shot 2016-11-21 at 9 23 10 pm" src="https://cloud.githubusercontent.com/assets/778171/20512165/72ea01d6-b032-11e6-926f-1d0b0299538f.png">  
  

__In process pages:__

<img width="1245" alt="screen shot 2016-11-21 at 9 40 39 pm" src="https://cloud.githubusercontent.com/assets/778171/20512290/440c2afa-b033-11e6-9ec1-1a5fe7300235.png">

  

__Missing email error:__
<img width="406" alt="screen shot 2016-11-21 at 9 23 29 pm" src="https://cloud.githubusercontent.com/assets/778171/20512164/6f94bd96-b032-11e6-916c-548478394a6d.png">


__Incomplete email error:__
<img width="424" alt="screen shot 2016-11-21 at 9 23 43 pm" src="https://cloud.githubusercontent.com/assets/778171/20512163/6c2fdda2-b032-11e6-9315-e026a538e2de.png">


__Server error:__

<img width="414" alt="screen shot 2016-11-21 at 9 24 00 pm" src="https://cloud.githubusercontent.com/assets/778171/20512158/691b49e4-b032-11e6-971c-90dee68a3daa.png">

__Server success:__

<img width="410" alt="screen shot 2016-11-21 at 9 24 55 pm" src="https://cloud.githubusercontent.com/assets/778171/20512155/66083dde-b032-11e6-9869-7befe0172c71.png">


## Notes

- This pulls in email signup & notification JS from cfgov-refresh, but adapts it slightly to use jQuery for DOM manipulation as a short-term solution until the project transitions away from IE8 support and adds the cfgov-refresh polyfills/dependencies. The code can be directly replaced with the cfgov-refresh versions once that transition takes place.

## Checklist
